### PR TITLE
feat(service-portal): add ADR and machine licenses to generic license service

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Adding API clients for the license service
  - ADR license
  - Machine license
* Generalizing the license service (it only worked for driving licenses

## Why

To implement the generalized license layer, and adding some licenses.

## Screenshots / Gifs

Screenshot 2022-05-18 at 11 27 34

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/license-service/src/lib/client/machine-license-client/genericMachineLicense.type.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export interface GenericMachineLicenseResponse {
```
Comment (osk) [inline libs/api/domains/license-service/src/lib/client/machine-license-client/genericMachineLicense.type.ts:1]: use exported `VinnuvelaDto` from service

File: libs/service-portal/core/src/utils/utils.ts [hunk 2]
```diff
@@ -54,3 +54,41 @@ export type DrivingLicenseType = {
   undirskrift: string
   svipting: Array<DrivingLicenseSuspended>
 }
+
+export type MachineLicenseCategoryType = {
```
Comment (osk) [inline libs/service-portal/core/src/utils/utils.ts:58]: use types from service

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          heiti: item.heiti,
+        })
+      })
+    })
+    return flattenedAdr
+  }
+
+  const parsedResponse = parseAdrLicenseResponse()
+
+  const data = [
```
Comment (osk) [inline libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts:40]: sync labels w design
https://www.figma.com/file/kzvPtwIM25MrBmoKOhEkji/%C3%8Dsland.is-%E2%80%93-App?node-id=2747%3A27519

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  license: AdrDto,
+): GenericUserLicensePayload | null => {
+  if (!license) return null
+
+  const parseAdrLicenseResponse = () => {
+    const { adrRettindi, ...rest } = license
+
+    const flattenedAdr: FlattenedAdrDto = { ...rest, adrRettindi: [] }
+
+    adrRettindi?.forEach((field) => {
```
Comment (osk) [inline libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts:18]: add comment about why

File: libs/api/domains/license-service/src/lib/client/machine-license-client/machineLicenseMappers.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  GenericUserLicensePayload,
+} from '../../licenceService.type'
+
+export const parseMachineLicensePayload = (
+  license: VinnuvelaDto,
+): GenericUserLicensePayload | null => {
+  if (!license) return null
+  const data = [
+    {
+      type: GenericLicenseDataFieldType.Value,
```
Comment (osk) [inline libs/api/domains/license-service/src/lib/client/machine-license-client/machineLicenseMappers.ts:13]: sync labels
https://www.figma.com/file/kzvPtwIM25MrBmoKOhEkji/%C3%8Dsland.is-%E2%80%93-App?node-id=2747%3A27519

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  async fetchLicense(user: User) {
+    let license: unknown
+
+    try {
+      license = await this.adrApi
+        .withMiddleware(new AuthMiddleware(user as Auth))
+        .getAdr()
+    } catch (e) {
+      console.log(e)
```
Comment (osk) [inline libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts:33]:
```suggestion
```
remove

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+  async getLicenseDetail(
+    user: User,
+  ): Promise<GenericLicenseUserdataExternal | null> {
+    return this.getLicense(user)
+  }
+  async getPkPassUrl(user: User): Promise<string | null> {
+    return null
+  }
+  async getPkPassQRCode(user: User): Promise<string | null> {
```
Comment (osk) [inline libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts:70]: disable lint rule

**Review by osk — APPROVED** [review]

**Review by peturgq — APPROVED** [review]

File: libs/clients/adr-and-machine-license/README.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# Clients ADR and Machine License
```
Comment (eirikurn) [inline libs/clients/adr-and-machine-license/README.md:1]: Should make title and navigation title consistent to other client libs.

**Review by eirikurn — APPROVED** [review]
Love it! Minor questions/comments.

File: libs/clients/adr-and-machine-license/README.md (hunk 8 above)
Comment (thorkellmani) [reply libs/clients/adr-and-machine-license/README.md:1]: Will do (later)

File: libs/clients/adr-and-machine-license/src/lib/adr-and-machine-license.config.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  name: 'AdrAndMachineLicenseClient',
+  schema,
+  load: (env) => ({
+    xRoadServicePath: env.required(
+      'XROAD_ADR_MACHINE_LICENSE_PATH',
+      'IS-DEV/GOV/10013/Vinnueftirlitid-Protected/rettindi-v1',
+    ),
+    fetch: {
+      timeout: 10000,
+      scope: ['@ver.is/rettindaskra'],
```
[the remark this answers is not in the document]
Comment (thorkellmani) [reply libs/clients/adr-and-machine-license/src/lib/adr-and-machine-license.config.ts:24]: Scope confirmed, merging now

**Pull request merged**